File: 1ºAno/Lista3/C/Casa/files/ex11.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main() {
    for (;;) {
        char* dados = (char*)malloc(100 * sizeof(char));
        long vals[2] = {0, 0};
        double tab = 0;
        for (;;) {
            printf ("Na mesma linha, entre, respectivamente, com os valores referentes a:\n\n1) Número da tabuada (podendo ser decimal ou inteiro)\n2) Índice inicial da tabuada\n3) Índice final da tabuada\nPara sair, entre com 0 em todos os valores\n\n");
            fgets (dados, 100, stdin);
            char *pt = strtok(dados, " ");
            tab = atoi(pt);
            int i = 0;
            while (pt != NULL) {
                pt = strtok(NULL, " ");
                vals[i] = atoi(pt);
                ++i;
            }
            if (vals[1] == 0 && tab != vals[0]) {
                printf ("*Número(s) inválidos... Tente novamente\n*");
            }
            else {
                break;
            }
        }
        if (tab == 0 && vals[0] == 0 && vals[1] == 0) {
            break;
        }
        if (vals[0] < vals[1]) {
            for (vals[0]; vals[0] <= vals[1]; ++vals[0]) {
                printf ("%.2lf X %ld = %.2lf\n", tab, vals[0], tab * vals[0]);
            }
        }
        else {
            for (vals[0]; vals[0] >= vals[1]; --vals[0]) {
                printf ("%.2lf X %ld = %.2lf\n", tab, vals[0], tab * vals[0]);
            }
        }
    }
    printf ("\nAdeus!");
    return 0;
}

File: 1ºAno/Lista3/Python/Casa/files/ex11.py
while True:
    while True:
        try:
            vals = input("Na mesma linha, entre respectivamente com os valores referentes a:\n\n1) Número para a tabuada\n2) Índice inicial da tabuada\n3) Índice final da tabuada\nPara sair, entre com 0 em todos os valores\n\n").split()
            vals[0], vals[1], vals[2] = float(vals[0]), int(vals[1]), int(vals[2])
            break
        except:
            print ("*Número(s) inválido(s)... Tente novamente*")
    if (vals[0] == vals[1] == vals[2] == 0):
        break
    if (vals[1] <= vals[2]):
        for vals[1] in range(vals[1], vals[2] + 1):
            print("{:.2f} X {} = {:.2f}" .format(vals[0], vals[1], vals[0] * vals[1]))
    else:
        for vals[1] in range(vals[1], vals[vals[2]] - 1, -1):
            print("{:.2f} X {} = {:.2f}" .format(vals[0], vals[1], vals[0] * vals[1]))
    print ()
print ("\nAdeus!")